src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -342,6 +346,116 @@ public final Text toText() {
         return this;
     }
 
+    /**
+     * Replaces a pattern in this text with a {@link Text}, preserving
+     * formatting where possible.
+     *
+     * <p>If {@code lossy} is true, this will replace the pattern if this finds
+     * it spanning multiple child {@link Text}s, resulting in a potential loss
+     * of formatting; if false, it will only replace the pattern where it is
+     * contained within one child {@link Text}, to preserve all formatting.</p>
+     * 
+     * <p>This method produces a copy with the specified replacements. The
+     * original is not modified.</p>
+     *
+     * @param oldValue The pattern to replace
+     * @param newValue The value to replace with
+     * @param lossy Whether lossy mode is enabled
+     * @return The resulting (copied) text
+     */
+    public Text replace(Pattern oldValue, Text newValue, boolean lossy) {
```

Comment:
Could you add a few comments to this method explaining what's going on? Will be nice for the next person who works on this.